src/vm.c

```diff
@@ -159,71 +159,54 @@ static val_t v7_create_object(struct v7 *v7, struct v7_object *proto) {
   }
   (void) v7;
   o->properties = NULL;
-  o->prototype = proto;
+  o->prototype = prototype == V7_UNDEFINED ? NULL : val_to_pointer(prototype);
```

Comment:
Or let's use `val_to_object` here and convert a V7_NULL into a NULL pointer in `val_to_object` (like we do the other way around in `v7_object_to_value`

(btw, at some point we should cleanup the `v7_xxx_to_value` names to `(y_)?xxx_to_val`